kensu/pyspark/spark_connector.py

```diff
@@ -509,54 +489,61 @@ def init_kensu_spark(
+        kensu_ingestion_token = extract_config_property('ingestion_token', None, kensu_ingestion_token, kw=kwargs, conf=kensu_conf)
+        report_to_file = extract_config_property('report_to_file', False, report_to_file, kw=kwargs, conf=kensu_conf, tpe=bool)
+        logs_dir_path = extract_config_property('logs_dir_path', None, logs_dir_path, kw=kwargs, conf=kensu_conf)
+        offline_file_name = extract_config_property('offline_file_name', 'kensu-offline.log', offline_file_name, kw=kwargs, conf=kensu_conf)
+
+        shutdown_timeout_sec = extract_config_property('shutdown_timeout_sec', 10 * 60, shutdown_timeout_sec, kw=kwargs, conf=kensu_conf, tpe=int)
+        kensu_api_verify_ssl = extract_config_property('api_verify_ssl', True, kensu_api_verify_ssl, kw=kwargs, conf=kensu_conf, tpe=bool)
+        enable_debugging = extract_config_property('enable_debugging', False, enable_debugging, kw=kwargs, conf=kensu_conf, tpe=bool)
+        debugging_log_level = extract_config_property('debugging_log_level', 'INFO', debugging_log_level, kw=kwargs, conf=kensu_conf)
+        debugging_include_spark_logs = extract_config_property('debugging_include_spark_logs', False, debugging_include_spark_logs, kw=kwargs, conf=kensu_conf, tpe=bool)
+
+        process_name = extract_config_property('process_name', None, process_name, kw=kwargs, conf=kensu_conf)
+        process_run_name = extract_config_property('process_run_name', None, process_run_name, kw=kwargs, conf=kensu_conf)
+        code_location = extract_config_property('code_location', None, code_location, kw=kwargs, conf=kensu_conf)
+        code_version = extract_config_property('code_version', None, code_version, kw=kwargs, conf=kensu_conf)
+        user_name = extract_config_property('user_name', None, user_name, kw=kwargs, conf=kensu_conf)
+        enable_entity_compaction = extract_config_property('enable_entity_compaction',True,enable_entity_compaction, kw=kwargs, conf=kensu_conf, tpe=bool)
+        environment = extract_config_property('environment',None,environment, kw=kwargs, conf=kensu_conf)
+        project_name = extract_config_property('project_name', None, project_name, kw=kwargs, conf=kensu_conf, tpe=list)  # type: list[str]
+        execution_timestamp = extract_config_property('execution_timestamp', None, execution_timestamp, kw=kwargs, conf=kensu_conf, tpe=int)
+
+        compute_stats = extract_config_property('compute_stats', True, compute_stats, kw=kwargs, conf=kensu_conf, tpe=bool)
+        compute_input_stats = extract_config_property('compute_input_stats', True, compute_input_stats, kw=kwargs, conf=kensu_conf, tpe=bool)
+        compute_output_stats = extract_config_property('compute_output_stats', True, compute_output_stats, kw=kwargs, conf=kensu_conf, tpe=bool)
+        input_stats_for_only_used_columns = extract_config_property('input_stats_for_only_used_columns',True,input_stats_for_only_used_columns, kw=kwargs, conf=kensu_conf, tpe=bool)
```

Comment:
should we handle defaults in Spark/Scala instead to reduce duplication (apply to most fields above/below but maybe not all)?

Reply:
@SammyEK what do you think? But also, I am running out of time.